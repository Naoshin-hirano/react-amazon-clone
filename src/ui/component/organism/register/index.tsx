import React from "react";
import "../../../../design/component/atoms/auth/register.css";
import { Link } from "@tanstack/react-location";

export const Register: React.FC<any> = (props) => {
    const { firstTitle } = props;
    console.log(firstTitle);

    return (
        <div className="register">
            <Link to="/">
                <img
                    alt="Amazon_logo"
                    className="register__logo"
                    src="https://upload.wikimedia.org/wikipedia/commons/thumb/a/a9/Amazon_logo.svg/1024px-Amazon_logo.svg.png"
                />
            </Link>
            <div className="register__container">
                <h1>新規登録</h1>
                <form>
                    <h5>メール</h5>
                    <input type="text" />

                    <h5>パスワード</h5>
                    <input type="password" />

                    <h5>パスワード（確認）</h5>
                    <input type="password" />

                    <button className="register_signInButton">新規登録</button>
                </form>

                <p>
                    登録することにより、Amazon FAKE CLONE's
                    に同意したことになります
                    使用および販売条件。当社のプライバシー通知を参照するか、
                    Cookies Notice および当社のインタレスト
                    ベース広告に関する通知。
                </p>

                <button className="register__registerButton">
                    Amazon Accountでログイン
                </button>
            </div>
        </div>
    );
};
