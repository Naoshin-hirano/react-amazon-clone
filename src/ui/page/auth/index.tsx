import React from "react";
import { Template } from "ui/component/template/auth";
import { useCreateProps } from "./presentation";

export const AuthPage: React.FC = () => {
    const props = useCreateProps();
    return <Template {...props}></Template>;
};
