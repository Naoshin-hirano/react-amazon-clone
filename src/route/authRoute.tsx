import { Route } from "@tanstack/react-location";
import React from "react";
import { LoginPage } from "ui/page/login";
import { RegisterPage } from "ui/page/register";

export const authRoute: Route[] = [
    {
        path: "/login",
        element: <LoginPage />,
    },
    {
        path: "/register",
        element: <RegisterPage />,
    },
];
