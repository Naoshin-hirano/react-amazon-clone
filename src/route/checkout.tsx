import { Route } from "@tanstack/react-location";
import React from "react";
// import { SocondSamplePage } from "ui/page/second";
import { CheckoutPage } from "ui/page/checkout";

export const checkoutRoute: Route[] = [
    {
        path: "/checkout",
        element: <CheckoutPage />,
    },
    // {
    //     path: "/resigter",
    //     element: <SocondSamplePage />,
    // },
];
