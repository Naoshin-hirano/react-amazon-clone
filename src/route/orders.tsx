import { Route } from "@tanstack/react-location";
import React from "react";
import { OrdersPage } from "ui/page/orders";

export const ordersRoute: Route[] = [
    {
        path: "/orders",
        element: <OrdersPage />,
    },
    // {
    //     path: "/resigter",
    //     element: <SocondSamplePage />,
    // },
];
