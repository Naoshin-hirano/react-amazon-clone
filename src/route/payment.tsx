import { Route } from "@tanstack/react-location";
import React from "react";
import { PaymentPage } from "ui/page/payment";

export const paymentRoute: Route[] = [
    {
        path: "/payment",
        element: <PaymentPage />,
    },
    // {
    //     path: "/resigter",
    //     element: <SocondSamplePage />,
    // },
];
