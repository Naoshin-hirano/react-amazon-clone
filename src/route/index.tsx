import { ReactLocation, Route } from "@tanstack/react-location";
import { history } from "./history";
import { authRoute } from "route/authRoute";
import React from "react";
import { TopPage } from "ui/page/top";

const indexRoute: Route[] = [
    {
        path: "/",
        element: <TopPage />,
    },
    {
        path: "/top",
        element: <TopPage />,
    },
];

export const routes: Route[] = [authRoute].reduce(
    (prevRoute, currRoute) => prevRoute.concat(currRoute),
    indexRoute
);

export const location = new ReactLocation({ history: history });
