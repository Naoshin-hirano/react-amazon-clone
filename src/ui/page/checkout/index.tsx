import React from "react";
import { Template } from "ui/component/template/checkout";
import { useCreateProps } from "./presentation";

export const CheckoutPage: React.FC = () => {
    const props = useCreateProps();
    return <Template {...props}></Template>;
};
