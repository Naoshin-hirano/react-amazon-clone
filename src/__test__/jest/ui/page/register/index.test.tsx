import { render } from "@testing-library/react";
import React from "react";
import { Router } from "@tanstack/react-location";
import { location, routes } from "../../../../../route";
import { RegisterPage } from "ui/page/register";

describe("RegisterPage", () => {
    beforeEach(() => {
        jest.clearAllMocks();
    });

    it("Register画面 snapshot確認", async () => {
        const { asFragment } = render(
            <Router routes={routes} location={location}>
                <RegisterPage />
            </Router>
        );
        expect(asFragment()).toMatchSnapshot("RegisterPage");
    });
});
