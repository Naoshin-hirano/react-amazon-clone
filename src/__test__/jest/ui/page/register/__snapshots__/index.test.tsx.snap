// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterPage Register画面 snapshot確認: RegisterPage 1`] = `
<DocumentFragment>
  <div>
    <div
      class="register"
    >
      <a
        class="active"
        href="/"
      >
        <img
          alt="Amazon_logo"
          class="register__logo"
          src="https://upload.wikimedia.org/wikipedia/commons/thumb/a/a9/Amazon_logo.svg/1024px-Amazon_logo.svg.png"
        />
      </a>
      <div
        class="register__container"
      >
        <h1>
          新規登録
        </h1>
        <form>
          <h5>
            メール
          </h5>
          <input
            type="text"
          />
          <h5>
            パスワード
          </h5>
          <input
            type="password"
          />
          <h5>
            パスワード（確認）
          </h5>
          <input
            type="password"
          />
          <button
            class="register_signInButton"
          >
            新規登録
          </button>
        </form>
        <p>
          登録することにより、Amazon FAKE CLONE's に同意したことになります 使用および販売条件。当社のプライバシー通知を参照するか、 Cookies Notice および当社のインタレスト ベース広告に関する通知。
        </p>
        <button
          class="register__registerButton"
        >
          Amazon Accountでログイン
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
